src/ExtraPackage.php

```diff
@@ -346,7 +342,7 @@ protected function mergeConstraints(Link $origin, Link $merge, PluginState $stat
         $newPrettyString = $merge->getConstraint()->getPrettyString();
 
         if ($state->isComposer1()) {
-            $constraintClass = 'Wikimedia\\Composer\\Merge\\MultiConstraint';
+            $constraintClass = 'Wikimedia\\Composer\\Merge\\V2\\MultiConstraint';
```

Comment:
Is it worth moving these to ::class (if they'll work) too?

Also, this looks kinda weird straight off; "if the state is composer (v) 1, use composer merge pluging V2 MultiConstraint, else use the one from Composer itself"

I know it's technically V2 of the plugin...

Reply:
I agree with you that the `V2` namespace feels weird. Sort of a necessary evil.

If only the Merge plugin had provided support for `setUpdateAllowList()` early on (before work on Composer 2 even began) then we _might_ have been able to avoid the renamespacing.

Reply:
Done, used ::class. With ::class it's not necessary to reference V2 in this line of code.